src/app/pages/template-data-specifications/template-data-specifications.component.ts

```diff
@@ -59,15 +72,19 @@ export class TemplateDataSpecificationsComponent implements OnInit {
     this.dataSpecification
       .listTemplates()
```

Comment:
I feel like the "Templates" page should have some sub-routes to determine whether you're looking at templates or community specifications. At the moment, if you select a community template from the list, view it, then go back, the list page reloads to its initial state of "Templates" which technically is not where you were.

For instance, if there were two routes added like:

- `/templates` - for the currated templates, and
- `/templates/community` - for the shared specifications

Then selecting the "Community" filter would redirect to the `/templates/community` route and `ngOnInit()` would decide which list to load based on the route parameters.

Reply:
Removed

Reply:
As discussed on a call, the idea was to avoid refreshing the page when clicking a dropdown, But I solved the issue about going back to the shared templates directly by adding a query parameter.